backend/controller/SequenceController.ts

```diff
@@ -0,0 +1,382 @@
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        return dbSequence;
+    }
+
+    /**
+     * Searches a slide of the sequence with given code. Throws an error if no sequence or slide could be found.
+     * @param code the code of the sequence
+     * @param slideIndex the index of the searched slide
+     * @returns the matching slide
+     */
+    public static async getSequenceSlideByCode(
+        code: string,
+        slideIndex: number
+    ): Promise<ISlide> {
+        const slide = await DBSlide.findOne({
+            where: { sequenceCode: code.toUpperCase(), order: slideIndex },
+        });
+        if (slide === null) {
+            throw new LoernwerkError(
+                'No matching Slide found',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        return slide;
+    }
+
+    /**
+     * stores the slides in the database
+     * @param slides the slides to store
+     */
+    private static async saveSlides(slides: ISlide[]): Promise<void> {
+        for (const s of slides) {
+            let slide = await DBSlide.findOneBy({ id: s.id });
+            if (slide === null) {
+                slide = new DBSlide();
+                slide.id = s.id;
+                slide.sequenceCode = s.sequenceCode;
+            }
+            slide.backgroundColor = s.backgroundColor;
+            slide.content = s.content;
+            slide.layout = s.layout;
+            slide.order = s.order;
+            await slide.save();
+        }
+    }
+
+    /**
+     * Generates a random code for a sequence.
+     * @returns  the generated sequence
+     */
+    private static generateCode(): string {
+        const codechar = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890';
+        const charactersLength = codechar.length;
+        let result = '';
+        for (let i = 0; i < length; i++) {
+            result += codechar.charAt(
+                Math.floor(Math.random() * charactersLength)
+            );
+        }
+        return result;
+    }
+
+    /**
+     * Checks if the given user id matches an user in the database
+     * @param userId the id of the user
+     * @returns true if the user exists
+     */
+    private static async isValidUser(userId: number): Promise<boolean> {
+        return (await DBUser.findOneBy({ id: userId })) !== null;
+    }
+    /**
+     * removing the given code from the given list. used on read/write access lists
+     * @param list the list
+     * @param code the code
+     * @returns the list wihtout the code
+     */
+    private static removeFromUserList(list: string[], code: string): string[] {
+        return list.splice(list.indexOf(code), 1);
+    }
+
+    /**
+     * updates users to fullfill a new access list
+     * @param oldAccessList the old access list
+     * @param newAccessList the new access list to fullfill
+     * @param code the code of the sequence
+     */
+    private static async updateSharedList(
```

Comment:
Maybe a better name? This sounds like we are updating a list that is shared with something